[doc] Add pulsar-client-go oauth2 authentication document (#7648)
apache/pulsar: site2/docs/client-libraries-go.md

    Authentication: NewAuthenticationTLS("my-cert.pem", "my-key.pem"),
}
```

## OAuth2 authentication

> In order to use [OAuth2 authentication](security-oauth2.md), you'll need to configure your client to do so:

Reviewer: ```suggestion
To use [OAuth2 authentication](security-oauth2.md), you'll need to configure your client to perform the following operations.
```

> Here's an example:

Reviewer: ```suggestion
This example shows how to configure Oauth2 authentication.
```
